### ISSUE TITLE
/etc/breakglass.authorized_keys: is a directory

### ISSUE THREAD

**Issue by bertmuthalaly** [gokrazy/breakglass#19]
This error surprised me - on macOS after setting up a new gokrazy device.

2024/06/06 12:44:12 gokrazy: attempt 5, starting ["/user/breakglass" "-authorized_keys=/etc/breakglass.authorized_keys"]
2024/06/06 12:44:12 breakglass.go:205: read /etc/breakglass.authorized_keys: is a directory

Trying to debug now.

**Comment by bertmuthalaly** [comment]
I worked around this by using `ExtraFileContents` instead, but I still don't know why breakglass.authorized_keys was a directory in the first place. In my gokrazy config dir breakglass.authorized_keys is definitely a file.

**Comment by stapelberg** [comment]
This is definitely unexpected.

I tried to reproduce it, but breakglass.authorized_keys is included as a file for me both on Linux and macOS.

Can you share the output of `ls -l` in your gokrazy instance directory?

Is the problem reproducible for you? Can you provide a minimal copy of the instance directory as a tar archive perhaps?

Another thing that would be interesting is the output of `gok overwrite --root /tmp/root.squashfs && unsquashfs -ll /tmp/root.squashfs`? (I installed the `unsquashfs` command from Homebrew using `brew install squashfs`)

**Comment by mennis** [comment]
I ran into what I believe to be this problem today.   The error is from breakglass is when loading the keys and ioutil.ReadFile on a directory returns an error `read <file>: is a directory`.  On the instance it was a directory. I wrote a new command to and installed it to verify this and then hit it from the web interface. 

This is the result of intentional and correct behavior in [packer](https://github.com/gokrazy/tools/blob/ee071a0b95256022d29cf5cbf9742c814229a6d7/internal/packer/packer.go#L581).
 
If the config variable ExtraFilePaths has a key in it that has an invalid value set for the path, packer will silently create that path as a directory for the user.   I had renamed my instance and had not updated the value assigned to 'breakglass.authorized_keys' . 

It _was_ surprising and confusing to me but I don't see a way to prevent this without commands having a way to validate their own configurations on build. Breakglass could, however, check the error or IsDir() and indicate
to the user through stderr that, perhaps, ExtraFilePaths has a bad value set for the authorized_keys file.
